services/etcd.go

```diff
@@ -140,3 +149,36 @@ func ReloadEtcdCluster(ctx context.Context, readyEtcdHosts []*hosts.Host, localC
 	}
 	return nil
 }
+
+func IsEtcdMember(ctx context.Context, etcdHost *hosts.Host, etcdHosts []*hosts.Host, localConnDialerFactory hosts.DialerFactory, cert, key []byte) (bool, error) {
+	var listErr error
+	peerURL := fmt.Sprintf("https://%s:2380", etcdHost.InternalAddress)
+	for _, host := range etcdHosts {
+		listErr = nil
+		if host.Address == etcdHost.Address {
+			continue
+		}
+		etcdClient, err := getEtcdClient(ctx, host, localConnDialerFactory, cert, key)
+		if err != nil {
+			logrus.Debugf("Failed to create etcd client for host [%s]: %v", host.Address, err)
+			continue
+		}
+		memAPI := etcdclient.NewMembersAPI(etcdClient)
+		members, err := memAPI.List(ctx)
+		if err != nil {
+			listErr = err
+			logrus.Debugf("Failed to list etcd cluster members [%s]: %v", etcdHost.Address, err)
+			continue
+		}
+		for _, member := range members {
+			if strings.Contains(member.PeerURLs[0], peerURL) {
+				logrus.Infof("[etcd] member [%s] is already part of the etcd cluster", etcdHost.Address)
+				return true, nil
+			}
+		}
+	}
+	if listErr != nil {
```

Comment:
should we return the contents of listError, so we can get the log for the reason in the caller stack? You can use `errors.Wrapf(err, "Failed to check if host [%s] is already part of the etcd cluster", etcdHost.Address)` construct.

Reply:
there is also a typo `par tof`